compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -434,12 +436,15 @@ ofInterestRule = do
                     let doc = formatScenarioResult world res
                     when (vr `Set.member` openVRs) $
                         sendEvent $ vrChangedNotification vr doc
+
         -- We don’t always have a scenario service (e.g., damlc compile)
         -- so only run scenarios if we have one.
         let shouldRunScenarios = isJust envScenarioService
-        _ <- parallel $
-            map (void . getDalf) (Set.toList scenarioFiles) <>
-            [runScenarios file | shouldRunScenarios, file <- Set.toList scenarioFiles]
+        let files = Set.toList scenarioFiles
+        let dalfActions = [(void . getDalf) f | f <- files]
+        let hlintActions = [use_ GetHlintDiagnostics f | f <- files]
```

Comment:
I would put the switch as to whether we enable hlint or not here similar to `shouldRunScenarios`. That has the nice side effect that it only controls the checks that we run by default and you can still trigger the HLint rule manually and you will get diagnostics.